### ISSUE TITLE
Twilio globally modifies Faraday's URL encoder

### ISSUE THREAD

**Issue by mguterl** [twilio/twilio-ruby#328]
https://github.com/twilio/twilio-ruby/blob/cc8880bf949e03882004bd72854d31c0bc1a41f1/lib/twilio-ruby/http/http_client.rb#L17

This changes the default behavior of Faraday for all users. It would be better to just provide the encoder where you build the requests for Twilio.

**Comment by robdimarco** [comment]
@mguterl thank you for finding us. We were chasing our tails looking for what was happening. I'm going to put in a Twilio support ticket referencing this as well.

**Comment by robdimarco** [comment]
FWIW, we monkey patched the `initialize` method to take out the offending line and it fixed our issues. I am not sure what functionality this might impact within the gem. From looking at the `git blame` it seems related to Marketplace APIs, which we are not using.

```
#
# Monkey patch to fix bug https://github.com/twilio/twilio-ruby/issues/328
#

# Make sure original file is loaded so we can monkey patch it and not be overwritten later
require 'twilio-ruby/http/http_client'

module Twilio
  module HTTP
    class Client
      def initialize(proxy_addr = nil, proxy_port = nil, proxy_user = nil, proxy_pass = nil, ssl_ca_file = nil)
        @proxy_addr = proxy_addr
        @proxy_port = proxy_port
        @proxy_user = proxy_user
        @proxy_pass = proxy_pass
        @ssl_ca_file = ssl_ca_file
        @adapter = Faraday.default_adapter

        #
        # HERE IS THE MONKEY IN THE MONKEY PATCh
        # comment out the line that sets the default params encoder
        # Faraday::Utils.default_params_encoder = Faraday::FlatParamsEncoder
      end
    end
  end
end
```

**Comment by mguterl** [comment]
This bug is particularly insidious because it is located inside of the initialize method. This only causes our code to fail after a Twilio job has been run. Our code worked fine from the console, but only failed when being run inside of Sidekiq after a job that touched Twilio. 

We didn't feel the need to monkey patch, this came about after upgrading from a much older version of the twilio gem. We just reverted back to the old version.

**Comment by robdimarco** [comment]
FWIW, Twilio Support has filed a ticket with their engineering team for this issue.

**Comment by efossier** [comment]
@robdimarco @mguterl Sorry for the trouble this caused you! We've corrected this in `5.1.1`.
